foreshadow/tests/test_transformers/test_internal.py

```diff
@@ -165,3 +165,43 @@ def test_drop_transformer_string_input():
     assert np.array_equal(
         x.values.ravel(), DropFeature().fit_transform(x).values.ravel()
     )
+
+
+def test_prepare_financial():
+    import numpy as np
+    import pandas as pd
+    from foreshadow.transformers.internals import PrepareFinancial
+
+    x = pd.DataFrame(
```

Comment:
Looking at the logic from PrepareFinancial, if a string contains a subsequence as a valid number, it should be able to keep that. For example, given string '2.3x', PrepareFinancial should be able to convert it to '2.3'. Is this interpretation right? Shall we test that behavior as well?

Reply:
I'll add that